Makefile

```diff
@@ -116,7 +116,7 @@ integration-test-binaries: generate ebpf
 	)
 
 docker-image:
-	docker build -t profiling-agent -f Dockerfile .
+	docker build --progress=plain -t profiling-agent -f Dockerfile --build-arg GOARCH=$(TARGET_ARCH) .
```

Comment:
`TARGET_ARCH` can be overridden when cross-compiling. I think the cross-compiler target arch is not intended here. But even better to not require the build-arg here, just determine the arch inside the `Dockerfile`.